roles/gitlab-runner/tasks/gitlab_runner.yml

```diff
@@ -1,26 +1,50 @@
-- name: pull the image
+- set_fact:
+    runner_config: "{{ default_runner_config|combine(item) }}"
+
+- name: get gitlab-runner-{{ runner_config.RUNNER_NAME }} run status
```

Comment:
I wish there were an elegant way to do this.

But if not i suppose we'll have to make do.